### ISSUE TITLE
Adding metadata to errors

### ISSUE THREAD

**Issue by kgnet88** [amantinband/error-or#25]
Hello,

one thing [FluentResults](https://github.com/altmann/FluentResults) has is the ability to add metadata. After switching to ErrorOr, I have to add the metadata inside `ProblemDetailsFactory`, which is very cumbersome... Due to this libraries implementation I have not find a elegant way to attach metadata to the error directly.

Do you know a way to add metadata in a similar way like in FluentResults or another generic way I can use to add metadata to `Error`. Or is it maybe even possible to add metadata to ErrorOr (maybe via generic `Error` instead of hardcoded one...).

Thanks in advance 😀

**Comment by kgnet88** [comment]
Different use cases:

1) environment metadata (helpful if the customer gives you only a screenshot of an error)
2) user context (session data, etc)
3) working with background APIs you may want to know which component is the real culprit (or if it was a resillience error)
4) compressed stacktrace, compressed binary log data

And this are only the ones on top of my head or ones I already have encountered.

**Comment by shanerogers** [comment]
@amantinband Good evening. I was looking at using ErrorOr<> in our solution. 
Currently, we use fluent validation with https://github.com/shanerogers/language-ext. 

More specifically Result<ValidationFailures> This is similar to your example (in docs) . 
However, I also need meta-data used in our problem json responses. So this is also a blocker for me.
Also worth mentioning. Is it possible to have 'PropertyName' as a field? So each error can have the following:

- ErrorCode
- PropertyName
- Description
- ErrorType
- Metadata

I also find the Numeric type isn't something I need. Also, wondering how that works with 'Custom' types.

**Comment by florentmx** [comment]
I personally would think that the easiest and most versatile solution would be to attach a simple dictionary to the error itself. Simple string as the key and an object as the value. My PR #37 could be used like this:

```
var metadata = new Dictionary<string, object>();
metadata.Add("id", 13947);

var error = Error.Failure("404", "entity not found", metadata);
```

Let me know what you think.

**Comment by vendre21** [comment]
I would really need this

**Comment by amantinband** [comment]
Added since I see many of you want this feature. Should be available in version 1.4.0 🙂
